_1327/documents/templates/editor.html

```diff
@@ -196,6 +231,38 @@
 							});
 						}
 					}]
+				},
+
+				{
+					name: "InternalGroup",
+					data: [{
+						name: "cmdAddInternalLink",
+						toggle: false,
+						title: "Add internal link",
+						icon: "glyphicon glyphicon-circle-arrow-right",
+						callback: function (e) {
+
+							var modalItem = $('#selectInternalLinkDestination');
+							modalItem.modal();
+
+							// set the click handler on the ok button that adds the markdown to the text
+							// and closes the modal
+							var setButton = $('#setInternalLinkDestination');
+							setButton.off('click');
+							setButton.on("click", function (event) {
+								modalItem.modal('hide');
+								// find the right spot to insert the text and create the correct markdown
+								var markdown = $('#selectInternalLinkModal').val();
+								if (markdown !== null) {
+									linkText = markdown.match(/\[(.*?)\]/)[1]; // we want the text inside the [ ]
+									selected = e.getSelection();
+									e.replaceSelection(markdown);
+									cursor = selected.start;
+									e.setSelection(cursor + 1, cursor + linkText.length + 1) // +1 because we do not want to select the starting [
```

Comment:
in firefox the selection seems to have no effect.
in chrome the text gets selected, but the edit field loses the focus once the modal is closed and the selection is not there anymore.